zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -178,6 +178,52 @@ def test_virsh_audit_action(self):
                     "The action failed: {}".format(action.data["message"]))
 
 
+class NovaCloudControllerActionTest(test_utils.OpenStackBaseTest):
+    """Run nova-cloud-controller specific tests.
+
+    Add this test class for new nova-cloud-controller action
+    to avoid breaking older version.
+    """
+
+    def test_sync_compute_az_action(self):
+        """Test sync-compute-availability-zones action."""
+        juju_units_az_map = {}
+        for unit in zaza.model.get_units('nova-compute',
+                                         model_name=self.model_name):
+            result = zaza.model.run_on_unit(unit.name,
+                                            'echo $JUJU_AVAILABILITY_ZONE',
+                                            model_name=self.model_name,
+                                            timeout=60)
+            self.assertEqual(int(result['Code']), 0)
+            juju_units_az_map[unit.public_address] = result['Stdout'].strip()
+            continue
+
+        session = openstack_utils.get_overcloud_keystone_session()
+        nova = openstack_utils.get_nova_session_client(session)
+        expected_output = ''
+        for hypervisor in nova.hypervisors.list():
+            az = juju_units_az_map.get(hypervisor.host_ip, None)
```

Comment:
Should we assert that it's not `None` everywhere? Because if have no AZ at all then I think the entire test is passing but not covering as much as we want.

Reply:
No, it shouldn't be checked. This actually highlighted for me that we're using the wrong setting in the availability zone in the nova-compute charm. I've raised bug https://bugs.launchpad.net/charm-nova-compute/+bug/1925412 and subsequently submitted https://review.opendev.org/c/openstack/charm-nova-compute/+/787470 to fix that.

To fix this, we need to be able to honor both maas and non-maas environments. I've got a branch with commits to handle this at https://github.com/wolsen/zaza-openstack-tests/tree/update-nova-tests/ - which are updates based on this branch/MR. I think it handles the scenario.

Reply:
@wolsen

Thank-you for raising (and solving) the issue with non-MAAS environments as well!

If you think it's fine, I'm more than happy to cherry-pick your fixes from:
https://github.com/wolsen/zaza-openstack-tests/tree/update-nova-tests/ 

And test this PR with those changes included.